Trimble.Modus.Components/Controls/TMCheckBox/TMCheckBox.cs

```diff
@@ -0,0 +1,188 @@
+
+        public bool IsDisabled
+        {
+            get => (bool)GetValue(IsDisabledProperty);
+            set => SetValue(IsDisabledProperty, value);
+        }
+        public CheckboxSize Size
+        {
+            get => (CheckboxSize)GetValue(SizeProperty);
+            set => SetValue(SizeProperty, value);
+        }
+        public string Text
+        {
+            get => (string)GetValue(TextProperty);
+            set => SetValue(TextProperty, value);
+        }
+        public bool IsChecked
+        {
+            get => (bool)GetValue(IsCheckedProperty);
+            set => SetValue(IsCheckedProperty, value);
+        }
+        public bool IsIndeterminate
+        {
+            get => (bool)GetValue(IsIndeterminateProperty);
+            set => SetValue(IsIndeterminateProperty, value);
+        }
+
+        public static readonly BindableProperty IsDisabledProperty =
+         BindableProperty.Create(
+             nameof(IsDisabled),
+             typeof(bool),
+             typeof(TMCheckBox),
+             true,
+             propertyChanged: OnIsDisabledChanged);
+
+        public static readonly BindableProperty IsIndeterminateProperty =
+          BindableProperty.Create(
+              nameof(IsIndeterminate),
+              typeof(bool),
+              typeof(TMCheckBox),
+              false,
+              propertyChanged: OnIsIndeterminateChanged);
+
+        private static void OnIsIndeterminateChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var customCheckboxView = (TMCheckBox)bindable;
+
+            if (customCheckboxView.IsIndeterminate) { 
+                customCheckboxView.IsChecked = false;
+                customCheckboxView._checkbox.Source = ImageSource.FromResource("Trimble.Modus.Components.Images.indeterminate_checkbox.png");
+
+            }
+            else
+            {
+                customCheckboxView.IsChecked = false;
+            }
+
+        }
+
+        public static readonly BindableProperty TextProperty =
+           BindableProperty.Create(
+               nameof(Text),
+               typeof(string),
+               typeof(TMCheckBox),
+               string.Empty,
+               propertyChanged: OnTextChanged);
+
+
+        public static readonly BindableProperty IsCheckedProperty =
+            BindableProperty.Create(
+                nameof(IsChecked),
+                typeof(bool),
+                typeof(TMCheckBox),
+                false,
+                propertyChanged: OnIsCheckedChanged);
+        public static readonly BindableProperty SizeProperty =
+                   BindableProperty.Create(
+                       nameof(Size),
+                       typeof(CheckboxSize),
+                       typeof(TMCheckBox),
+                       CheckboxSize.Default,
+                       propertyChanged: OnSizeChanged);
+
+        private static void OnSizeChanged(BindableObject bindable, object oldValue, object newValue)
```

Comment:
move this private method below the constructor

Reply:
Formatted